lib/src/bindings.dart

```diff
@@ -217,3 +225,36 @@ abstract class DeltaDocumentSnapshot implements admin.DocumentSnapshot {
   /// Gets the value for a given key.
   external dynamic get(key);
 }
+
+@JS()
+@anonymous
+abstract class PubsubFunctions {
+  /// Registers a function that triggers on Pubsub write events to
+  /// the [topic].
+  external TopicBuilder topic(String topic);
+}
+
+/// The Pubsub topic builder interface.
+@JS()
+@anonymous
+abstract class TopicBuilder {
+  /// Event handler that fires every time an event is public in Pubsub.
+  external CloudFunction onPublish(void handler(Message message));
```

Comment:
Relevant to my other comment, handler should probably expect `Event` containing `Message`.

Note though that in JS bindings I don't have subclasses for events and just use `Event<DeltaDocumentSnapshot>` for Firestore, for instance. This is for simplicity and also to closer match JS docs.